src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -0,0 +1,846 @@
+                if (_filterText.Length == 0)
+                {
+                    // Without filterText, all items are equally good match (w.r.t to the empty filterText), so we have to consider all of them.
+                    inferiorItemIndex = items.Count;
+                }
+                else
+                {
+                    // Because the items are sorted based on pattern-matching score, the selectedIndex is in the middle of a range of
+                    // -- as far as the pattern matcher is concerned -- equivalent items (items with identical PatternMatch.Kind and IsCaseSensitive).
+                    // Find the last items in the range and use that to limit the items searched for from the defaults list.          
+                    var selectedItemMatch = items[intialSelection.SelectedItemIndex].PatternMatch;
+
+                    if (!selectedItemMatch.HasValue)
+                        return intialSelection;
+
+                    inferiorItemIndex = intialSelection.SelectedItemIndex;
+                    while (++inferiorItemIndex < items.Count)
+                    {
+                        var itemMatch = items[inferiorItemIndex].PatternMatch;
+                        if (!itemMatch.HasValue
+                            || itemMatch.Value.Kind != selectedItemMatch.Value.Kind
+                            || itemMatch.Value.IsCaseSensitive != selectedItemMatch.Value.IsCaseSensitive)
+                        {
+                            break;
+                        }
+                    }
+                }
+
+                foreach (var defaultText in _data.Defaults)
+                {
+                    // The range includes all items that are as good of a match as what we intially selected (and in descending order of matching score)
+                    // so we just need to search for the first item that matches the suggested default.
+                    for (var i = 0; i < inferiorItemIndex; ++i)
+                    {
+                        if (items[i].RoslynCompletionItem.DisplayText == defaultText)
+                            return intialSelection with { SelectedItemIndex = i };
+                    }
+                }
+
+                return intialSelection;
+            }
+
+            private sealed class PooledListForMatchResult : IDisposable
+            {
+                // We might need to handle large amount of items with import completion enabled,
+                // so use a dedicated pool to minimize/avoid array allocations (especially in LOH)
+                // Set the size of pool to 1 because we don't expect UpdateCompletionListAsync to be
+                // called concurrently, which essentially makes the pooled list a singleton,
+                // but we still use ObjectPool for concurrency handling just to be robust.
+                private static readonly ObjectPool<List<MatchResult<VSCompletionItem>>> s_listOfMatchResultPool = new(factory: () => new(), size: 1);
```

Comment:
fwiw, you can just remove hte 'size'.  if we don't have concurrency, we'll always just give/return the same item.  and, if we do, then we still pool, which seems ok.